### PR TITLE
Upgrade of tests requirements + resolving current deprecation warnings

### PR DESCRIPTION
**Pull request by dom-nie**
Upgrade to latest versions of packages for tests with all necessary changes.
Adding also ignore for DeprecationWarning, since third party packages like requests show some warnings, which can't be affected from this project perspective.

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[pytest]
+filterwarnings =
+    ignore::DeprecationWarning
```
Comment (bastelfreak) [inline pytest.ini]: please add the missing newline at the end of the file

**Comment by bastelfreak** [comment]
@dom-nie thanks for the PR. Can you take a look at the failing travis jobs?

**Comment by dom-nie** [comment]
@bastelfreak I went ahead and  opened #531 just to resolve this which I believe is best solution.
If you think we should keep Python2 let me know and I will upgrade this branch accordingly.

File: pytest.ini (hunk 1 above)
Comment (dom-nie) [reply pytest.ini]: Added.

**Comment by bastelfreak** [comment]
let's rip it out in the other PR and afterwards rebase this one.

**Pull request merged**